Documentation/Porting.md

```diff
@@ -0,0 +1,251 @@
+# Porting to new platforms
+
+<!--
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2024 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+When the Swift toolchain is ported to a new platform, it is necessary to port
+Swift Testing as well. This document contains various information, trivia, and
+deep wisdoms about porting Swift Testing.
+
+> [!NOTE]
+> This document uses Classic Mac OS ("Classic") as an example target platform.
+> In this hypothetical scenario, we assume that the Swift compiler identifies
+> Classic with `os(Classic)` and that the C++ compiler identifies it with
+> `defined(macintosh)`. Other platforms would be identified differently.
+
+## Getting started
+
+Before you start the porting process, make sure you are very familiar with Swift
+and C++ as well as the C standard library and platform SDK for your target
+platform.
+
+Your first task when porting Swift Testing is ensuring that it builds.
+
+We've made an effort to ensure that as much of our code as possible is
+platform-agnostic. When building the toolchain for a new platform, you will
+hopefully find that Swift Testing builds out-of-the-box with few, if any,
+errors.
+
+> [!NOTE]
+> Swift Testing relies on the Swift [standard library](https://github.com/swiftlang/swift),
+> Swift macros (including the [swift-syntax](https://github.com/swiftlang/swift-syntax) package),
+> and [Foundation](https://github.com/apple/swift-foundation). These components
```

Comment:
All our Foundation uses are guarded by `#if canImport(Foundation)`. Should Foundation rather be described as an optional dependency here?

Reply:
Eh, I don't think so. I would expect most people attempting a port are doing so to a system where Foundation will be properly available. (Embedded Swift is a different animal entirely.) The fact that we compile without it is a quirk of our build process, not really something we want to support.